problemtools/run/__init__.py

```diff
@@ -62,7 +62,7 @@ def find_programs(path, pattern='.*', language_config=None, work_dir=None,
 
 
 def get_program(path, language_config=None, work_dir=None, include_dir=None,
-                allow_validation_script=False):
+                allow_validation_script=False) -> Program|None:
```

Comment:
Perhaps use
```suggestion
                allow_validation_script=False) -> Optional[Program]:
```
instead?

Reply:
I think `|None` is the more modern version (added in 3.10)? It matches what the rest of the code base uses, too. (I do agree personally that `|None` is a bit ugly.)

Reply:
I was unaware of that and have now read through the following were this is debated: https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/44

You can ignore all my suggestions then, I have no strong preference on this at all.